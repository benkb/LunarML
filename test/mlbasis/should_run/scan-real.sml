fun printOptionReal NONE = print "NONE\n"
  | printOptionReal (SOME x) = print ("SOME " ^ Real.toString x ^ "\n")
val () = List.app (fn s => (print ("Real.fromString \"" ^ String.toString s ^ "\" = "); printOptionReal (Real.fromString s)))
                  [ ""
                  , "0"
                  , "1.0"
                  , "inf"
                  , "+inf"
                  , " ~INF"
                  , "\t-Inf"
                  , "-NaN"
                  , "1z"
                  , "~.5e0"
                  , "~5e~1"
                  , "~5E-1"
                  , "+5E"
                  , "5E1000"
                  ];
