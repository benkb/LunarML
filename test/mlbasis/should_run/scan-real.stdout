Real.fromString "" = NONE
Real.fromString "0" = SOME 0
Real.fromString "1.0" = SOME 1
Real.fromString "inf" = SOME inf
Real.fromString "+inf" = SOME inf
Real.fromString " ~INF" = SOME ~inf
Real.fromString "\t-Inf" = SOME ~inf
Real.fromString "-NaN" = SOME nan
Real.fromString "1z" = SOME 1
Real.fromString "~.5e0" = SOME ~0.5
Real.fromString "~5e~1" = SOME ~0.5
Real.fromString "~5E-1" = SOME ~0.5
Real.fromString "+5E" = SOME 5
Real.fromString "5E1000" = SOME inf
